ingress/controllers/nginx/README.md

```diff
@@ -2,19 +2,27 @@
 
 This is a nginx Ingress controller that uses [ConfigMap](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/configmap.md) to store the nginx configuration. See [Ingress controller documentation](../README.md) for details on how it works.
 
+## Contents
+* [Requirements](#what-it-provides)
+* [Dry running](#dry-running-the-ingress-controller)
+* [Deployment](#deployment)
+* [HTTP](#http)
+* [HTTPS](#https)
+  * [HTTPS enforcement](#server-side-https-enforcement)
+  * [HSTS](#http-strict-transport-security)
+* [TCP Services](#exposing-tcp-services)
+* [UDP Services](#exposing-udp-services)
+* [Proxy Protocol](#proxy-protocol)
```

Comment:
consider replacing this with a section called source ip that contains: xforwarded for, proxy protocol, and the normal source ip that you'd get with (normal pod (no host port), host port, net=host).